### ISSUE TITLE
clj-http 0.9.2 vs. Clojure 1.7.0

### ISSUE THREAD

**Issue by xsc** [technomancy/leiningen#1959]
Leiningen's `clj-http` dependency (0.9.2) seems to be not completely compatible with the Clojure version Leiningen provides. Plugins using `clj-http` seem to be unable to override the HTTP library's version.

---

__Details__

Create a new plugin project:

```
$ lein new plugin tester && cd tester
```

Start a REPL and require `clj-http.client`:

```
$ lein repl
user=> (require 'clj-http.client)
WARNING: update already refers to: #'clojure.core/update in namespace: clj-http.client, being replaced by: #'clj-http.client/update
nil
```

Explictly adding `:dependencies [[clj-http "2.0.0"]]` to `project.clj` does not change the outcome.

---

__Environment__

```
$ lein version
Leiningen 2.5.2 on Java 1.8.0_45 Java HotSpot(TM) 64-Bit Server VM
```

All plugins were removed from `$LEIN_HOME/profiles.clj`.

Problem encountered while investigating xsc/lein-ancient#58.

**Comment by xsc** [comment]
I assume that this can be resolved by upgrading `clj-http`. Can I also assume that there is a reason why it's still at 0.9.2?

**Comment by hyPiRion** [comment]
Bumping to 2.0.0 would move from the Apache http stuff to 4.3 to 4.5, which is a big jump I don't want to take before properly testing it. It was something to take into consideration for ffa700f94635175f933108633878c1d8438f7185, so I'd like to ensure that nothing breaks upstream before jumping to 2.0.0

That said, I'd like to bump stale dependencies, so I'd love some testing on what happens if we bump this to 2.0.0 without doing anything. Preferably on some of the stuff that is using wagon things.

**Comment by gigafone** [comment]
+1 This would be nice to fix. I'll help test if you like.

**Comment by malchmih** [comment]
+1

**Comment by jasonjckn** [comment]
+1

**Comment by hyPiRion** [comment]
Would anyone interested in getting this one fixed test out #2032? We need to verify that we don't break dependency and deployment tasks.

**Comment by haroldcarr** [comment]
What is the status of this fix? 

 I am still seeing the problem with lein 2.5.3 where my project uses clj-http 2.0.0

**Comment by hyPiRion** [comment]
Oh whoops, this is actually solved by #2032. It'll be available in 2.6.0 when it comes out.

**Comment by haroldcarr** [comment]
Great - any ETA on 2.6.0?

thanks

**Comment by hyPiRion** [comment]
https://github.com/technomancy/leiningen/milestones/2.6.0 is the milestone.

The important pieces that has to get fixed are #1966, #2033 and #1957. The others are effectively solved but needs some attention and care.

I may also want to look into fixing some issues related to mirrors for Clojars, but that would only happen if it's easy to do.
